required/latex-lab/latex-lab-sec.dtx

```diff
@@ -359,11 +359,12 @@
 %
 % The commands \cs{mainmatter}, \cs{backmatter}, \cs{frontmatter} and
 % \cs{appendix} close all \texttt{Sect} and \texttt{Part} structures.
+% \changes{v0.84g}{2025/01/05}{ensure that paragraph is ended (tagging/777)}
 %    \begin{macrocode}
-\AddToHook{cmd/frontmatter/before}{\@@_sec_end:n{-10}}
-\AddToHook{cmd/mainmatter/before} {\@@_sec_end:n{-10}}
-\AddToHook{cmd/backmatter/before} {\@@_sec_end:n{-10}}
-\AddToHook{cmd/appendix/before}   {\@@_sec_end:n{-10}}
+\AddToHook{cmd/frontmatter/before}{\par\@@_sec_end:n{-10}}
+\AddToHook{cmd/mainmatter/before} {\par\@@_sec_end:n{-10}}
+\AddToHook{cmd/backmatter/before} {\par\@@_sec_end:n{-10}}
+\AddToHook{cmd/appendix/before}   {\par\@@_sec_end:n{-10}}
```

Comment:
Probably better would be to move the \par command into those comands eventually (to ensure that they always behave identically (with or without tagging).

Also (not checked) but isn't `\@@_sec_end:n{-10}` a related command and should be in a socket rather than a hook? 

Doesn't have to be done now, but there should be a TODO if this is something we should cleanup at some point.

Reply:
I'm not sure if a \par inside `\@@_sec_end:n` is always wanted that's why I added if here explicitly (In the other cases the command is currently is used inside the sectioning commands in vmode, so a \par inside the command is not needed).

I'm not sure what you mean by sockets. \mainmatter etc are defined by the classes, so we can not simply redefine them to add a socket but have to use a generic hook. The problem was that the hook is before the \clearpage and so not necessarly in vmode.